src/io.c

```diff
@@ -690,92 +527,166 @@ int EmbedGenerateCookie(WOLFSSL* ssl, byte *buf, int sz, void *ctx)
 #endif /* WOLFSSL_SESSION_EXPORT */
 #endif /* WOLFSSL_DTLS */
 
-#ifdef HAVE_OCSP
+#endif /* USE_WOLFSSL_IO */
+
+
+#if defined(USE_WOLFSSL_IO)
+
+#ifndef DEFAULT_TIMEOUT_SEC
+    #define DEFAULT_TIMEOUT_SEC 0 /* no timeout */
+#endif
+
+#ifdef HAVE_IO_TIMEOUT
+    static int io_timeout_sec = DEFAULT_TIMEOUT_SEC;
+#else
+    #define io_timeout_sec DEFAULT_TIMEOUT_SEC
+#endif
+
 
-#include <stdlib.h>   /* atoi() */
+void wolfIO_SetTimeout(int to_sec)
+{
+#ifdef HAVE_IO_TIMEOUT
+    io_timeout_sec = to_sec;
+#else
+    (void)to_sec;
+#endif
+}
 
+int wolfIO_SetBlockingMode(SOCKET_T sockfd, int non_blocking)
+{
+    int ret = 0;
+
+#ifdef USE_WINDOWS_API
+    unsigned long blocking = non_blocking;
+    ret = ioctlsocket(sockfd, FIONBIO, &blocking);
+    if (ret == SOCKET_ERROR)
+        ret = -1;
+#else
+    ret = fcntl(sockfd, F_GETFL, 0);
```

Comment:
On non Windows platforms we can't assume fcntl() and O_NONBLOCK are going to be available.